### ISSUE TITLE
[2.7][3.6] test_socketserver ForkingMixIn tests leaks child processes on Python 3.6

### ISSUE THREAD

**Issue by vstinner** [python/cpython#75774]
BPO | [31593](https://bugs.python.org/issue31593)
--- | :---
Nosy | @vstinner
PRs | <li>python/cpython#3766</li><li>python/cpython#3786</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-09-27.10:24:21.543>
created_at = <Date 2017-09-26.13:47:40.624>
labels = []
title = '[2.7][3.6] test_socketserver ForkingMixIn tests leaks child processes on Python 3.6'
updated_at = <Date 2017-09-27.10:24:21.542>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2017-09-27.10:24:21.542>
actor = 'vstinner'
assignee = 'none'
closed = True
closed_date = <Date 2017-09-27.10:24:21.543>
closer = 'vstinner'
components = []
creation = <Date 2017-09-26.13:47:40.624>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 31593
keywords = ['patch']
message_count = 3.0
messages = ['303040', '303115', '303120']
nosy_count = 1.0
nosy_names = ['vstinner']
pr_nums = ['3766', '3786']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31593'
versions = ['Python 2.7', 'Python 3.6']
```

</p></details>

**Comment by vstinner** [comment]
The bpo-31151 (socketserver.ForkingMixIn.server_close() leaks zombie processes) was fixed in the master branch, but 2.7 and 3.6 branches were not changed.

I proposed to modify test_socketserver to explicitly waits until child processes complete. See attached PR.

Example of leaked child processes the on "x86 Gentoo Refleaks 3.6" buildbot:

http://buildbot.python.org/all/builders/x86%20Gentoo%20Refleaks%203.6/builds/102/steps/test/logs/stdio

0:26:11 load avg: 5.72 [ 62/405] test_socketserver passed -- running: test_multiprocessing_forkserver (594 sec)
beginning 6 repetitions
123456
Warning -- reap_children() reaped child process 6891
.Warning -- reap_children() reaped child process 6976
Warning -- reap_children() reaped child process 6981
.Warning -- reap_children() reaped child process 7059
.Warning -- reap_children() reaped child process 7136
..Warning -- reap_children() reaped child process 7306
.
